### ISSUE TITLE
first(named_vector) removes names

### ISSUE THREAD

**Issue by franknarf1** [Rdatatable/data.table#2487]
``` r
library(data.table)
x = setNames(, "A")
x[1]
#>   A 
#> "A"
first(x)
#> [1] "A"
```

In `example(first)`, it suggests `first(x)` and `x[1]` should be interchangeable, but it doesn't hold here.

**Comment by MichaelChirico** [comment]
To be clear, you aren't requesting that `first(x)` keeps the names, just that the documentation should clarify this distinction, right?

**Comment by franknarf1** [comment]
@MichaelChirico Yeah, a very minor request, but it bit me when I was using `lapply` and then accessing the results by name. Maybe `first`/`last` also strips other attributes; I didn't check and don't know the internals, so I'm not sure exactly what the doc should say about it.

**Comment by Henrik-P** [comment]
Somewhat related: [Documentation: first(x) # [not always the] same as x[1]](https://github.com/Rdatatable/data.table/issues/2002)

**Comment by jangorecki** [comment]
related https://github.com/Rdatatable/data.table/pull/4065#issuecomment-557763540
